viz/cmd/tap.go

```diff
@@ -353,83 +361,77 @@ func renderTapEvent(event *tapPb.TapEvent, resource string) string {
 		tls,
 	)
 
-	// If `resource` is non-empty, then
-	resources := ""
-	if resource != "" {
-		resources = fmt.Sprintf(
-			"%s%s%s",
-			src.formatResource(resource),
-			dst.formatResource(resource),
-			routeLabels(event),
-		)
-	}
-
+	var formattedEvent string
 	switch ev := event.GetHttp().GetEvent().(type) {
 	case *tapPb.TapEvent_Http_RequestInit_:
-		return fmt.Sprintf("req id=%d:%d %s :method=%s :authority=%s :path=%s%s",
+		formattedEvent = fmt.Sprintf("req id=%d:%d %s :method=%s :authority=%s :path=%s",
 			ev.RequestInit.GetId().GetBase(),
 			ev.RequestInit.GetId().GetStream(),
 			flow,
 			vizutil.HTTPMethodToString(ev.RequestInit.GetMethod()),
 			ev.RequestInit.GetAuthority(),
 			ev.RequestInit.GetPath(),
-			resources,
 		)
 
 	case *tapPb.TapEvent_Http_ResponseInit_:
-		return fmt.Sprintf("rsp id=%d:%d %s :status=%d latency=%dµs%s",
+		formattedEvent = fmt.Sprintf("rsp id=%d:%d %s :status=%d latency=%dµs",
 			ev.ResponseInit.GetId().GetBase(),
 			ev.ResponseInit.GetId().GetStream(),
 			flow,
 			ev.ResponseInit.GetHttpStatus(),
 			ev.ResponseInit.GetSinceRequestInit().GetNanos()/1000,
-			resources,
 		)
 
 	case *tapPb.TapEvent_Http_ResponseEnd_:
 		switch eos := ev.ResponseEnd.GetEos().GetEnd().(type) {
 		case *metricsPb.Eos_GrpcStatusCode:
-			return fmt.Sprintf(
-				"end id=%d:%d %s grpc-status=%s duration=%dµs response-length=%dB%s",
+			formattedEvent = fmt.Sprintf(
+				"end id=%d:%d %s grpc-status=%s duration=%dµs response-length=%dB",
 				ev.ResponseEnd.GetId().GetBase(),
 				ev.ResponseEnd.GetId().GetStream(),
 				flow,
 				codes.Code(eos.GrpcStatusCode),
 				ev.ResponseEnd.GetSinceResponseInit().GetNanos()/1000,
 				ev.ResponseEnd.GetResponseBytes(),
-				resources,
 			)
 
 		case *metricsPb.Eos_ResetErrorCode:
-			return fmt.Sprintf(
-				"end id=%d:%d %s reset-error=%+v duration=%dµs response-length=%dB%s",
+			formattedEvent = fmt.Sprintf(
+				"end id=%d:%d %s reset-error=%+v duration=%dµs response-length=%dB",
 				ev.ResponseEnd.GetId().GetBase(),
 				ev.ResponseEnd.GetId().GetStream(),
 				flow,
 				eos.ResetErrorCode,
 				ev.ResponseEnd.GetSinceResponseInit().GetNanos()/1000,
 				ev.ResponseEnd.GetResponseBytes(),
-				resources,
 			)
 
 		default:
-			return fmt.Sprintf("end id=%d:%d %s duration=%dµs response-length=%dB%s",
+			formattedEvent = fmt.Sprintf("end id=%d:%d %s duration=%dµs response-length=%dB",
 				ev.ResponseEnd.GetId().GetBase(),
 				ev.ResponseEnd.GetId().GetStream(),
 				flow,
 				ev.ResponseEnd.GetSinceResponseInit().GetNanos()/1000,
 				ev.ResponseEnd.GetResponseBytes(),
-				resources,
 			)
 		}
 
 	default:
-		return fmt.Sprintf("unknown %s", flow)
+		formattedEvent = fmt.Sprintf("unknown %s", flow)
 	}
+
+	if wide {
+		out := []string{formattedEvent}
+		out = append(out, src.formatResource()...)
+		out = append(out, dst.formatResource()...)
+		out = append(out, routeLabels(event)...)
+		return strings.Join(out, " ")
+	}
+	return formattedEvent
```

Comment:
I'd be inclined to separate this into another function so that we can use `return` to return formatted events and then decorate it in a wrapper; and then we don't need the `wide` flag:

Maybe:

```go
func renderTapEventWide(event *tapPb.TapEvent) string {
    out := []string{formatTapEvent(event)}
	out = append(out, src(event).formatResource()...)
	out = append(out, dst(event).formatResource()...)
	out = append(out, routeLabels(event)...)
	return strings.Join(out, " ")
}

func renderTapEventNarrow(event *tapPb.TapEvent) string {
    return formatTapEvent(event)
}

func formatTapEvent(event *tapPb.TapEvent) string {
    // ...
}
```